app/controllers/api/motorcycles_controller.rb

```diff
@@ -1,2 +1,44 @@
 class Api::MotorcyclesController < ApplicationController
+  before_action :set_motorcycle, only: [:show, :update, :destroy]
+
+  def index
+    @motorcycles = Motorcycle.all
+    render json: @motorcycles
+  end
+
+  def show
+    render json: @motorcycle
+  end
```

Comment:
Which motorcycle are we returning here, I think we first need to find which motor is being requested, do
`motor = Motorcycle.find_by(id:params[:id]) #then`
```
if not  motor.nil?
#statemetns
end
```
- And Add messages and status code in the response body

Reply:
There is a private set_motorcycle method that retrieves the motorcycle with the given id parameter from the database and sets it to the @motorcycle instance variable. And Rails handles response status codes and messages automatically.